Orchestrator.py

```diff
@@ -102,27 +109,43 @@ def _sendDownstreamCommands(self):
             sender = self,
         )
 
-    def receive(self,frame):
+    def _updateMovement(self, source):
```

Comment:
per above, rename the param.
Also, let's make things more explicit:
- rename to `_pickNewMovement(dotBotId)` and return (heading,speed) which you assign to the DotBotview in the caller
- do NOT call `computeCurrentPosition()` in this function, all it does is picking a new movement
- you need to call `computeCurrentPosition()` in the caller